plugins/mongo_db_plugin/include/eosio/mongo_db_plugin/bson.hpp

```diff
@@ -0,0 +1,248 @@
+         c.append(v.as_double());
+         return;
+      case fc::variant::bool_type:
+         c.append(v.as_bool());
+         return;
+      case fc::variant::string_type: {
+         c.append(v.as_string());
+         return;
+      }
+      case fc::variant::blob_type: {
+         bsoncxx::types::b_binary bin;
+         bin.sub_type = bsoncxx::binary_sub_type::k_binary;
+         bin.size = v.as_blob().data.size();
+         bin.bytes = reinterpret_cast<uint8_t*>(&(*v.as_blob().data.begin()));
+         c.append(bin);
+         return;
+      }
+      case fc::variant::array_type: {
+         const fc::variants& a = v.get_array();
+         bsoncxx::builder::core sub(true);
+         to_bson(a, sub);
+         c.append(sub.extract_array());
+         return;
+      }
+      case fc::variant::object_type: {
+         const fc::variant_object& o = v.get_object();
+         if (o.size() == 1) {
+            const auto value = o.begin()->value();
+            if (o.begin()->key().compare("$oid") == 0) {
+               if (value.get_type() == fc::variant::string_type
+                  && value.as_string().size() == 12 * 2) {
+                  bsoncxx::oid oid(value.as_string());
+                  c.append(oid);
+                  break;
+               }
+            }
+            else if (o.begin()->key().compare("$date") == 0) {
+               if (value.get_type() == fc::variant::int64_type) {
+                  bsoncxx::types::b_date date(std::chrono::milliseconds(value.as_int64()));
+                  c.append(date);
+                  break;
+               }
+               else if (value.get_type() == fc::variant::object_type) {
+                  const fc::variant_object& obj = value.get_object();
+                  if (obj.size() == 1) {
+                     auto number = obj.begin();
+                     if (number->key().compare("$numberLong") == 0) {
+                        bsoncxx::types::b_date date(std::chrono::milliseconds(number->value().as_int64()));
+                        c.append(date);
+                        break;
+                     }
+                  }
+               }
+            }
+            else if (o.begin()->key().compare("$timestamp") == 0) {
+               if (value.get_type() == fc::variant::object_type) {
+                  const fc::variant_object& obj = value.get_object();
+                  if (obj.size() == 2) {
+                     auto t = obj.begin();
+                     auto i = t;
+                     ++i;
+                     if (t->key().compare("t") == 0 && i->key().compare("i") == 0) {
+                        bsoncxx::types::b_timestamp ts;
+                        ts.timestamp = static_cast<uint32_t>(t->value().as_uint64());
+                        ts.increment = static_cast<uint32_t>(i->value().as_uint64());
+                        c.append(ts);
+                        break;
+                     }
+                  }
+               }
```

Comment:
Add an `else` that logs an error.

Reply:
the special handling is optional, not necessarily an error.